src/app/right-bar-creators/right-bar-creators-leaderboard/right-bar-creators-leaderboard.component.ts

```diff
@@ -1,57 +1,84 @@
-import { Component, OnInit } from "@angular/core";
+import { Component, Input, OnInit } from "@angular/core";
 import { GlobalVarsService } from "../../global-vars.service";
 import { ActivatedRoute, Router } from "@angular/router";
 import { BackendApiService } from "../../backend-api.service";
+import { RightBarCreatorsComponent } from "../right-bar-creators.component";
+import { HttpClient } from "@angular/common/http";
+import {
+  LeaderboardResponse,
+  LeaderboardToDataAttribute,
+  PulseLeaderboardType,
+  PulseService,
+} from "../../../lib/services/pulse/pulse-service";
 
 @Component({
   selector: "right-bar-creators-leaderboard",
   templateUrl: "./right-bar-creators-leaderboard.component.html",
   styleUrls: ["./right-bar-creators-leaderboard.component.scss"],
 })
 export class RightBarCreatorsLeaderboardComponent implements OnInit {
+  @Input() activeTab: string;
+
   static MAX_PROFILE_ENTRIES = 10;
+  RightBarCreatorsComponent = RightBarCreatorsComponent;
+
+  topGainerProfiles: any[] = [];
+
+  topDiamondProfiles: any[] = [];
 
   constructor(
     public globalVars: GlobalVarsService,
     private route: ActivatedRoute,
     private _router: Router,
-    private backendApi: BackendApiService
+    private backendApi: BackendApiService,
+    private httpClient: HttpClient
   ) {}
 
   ngOnInit() {
     if (this.globalVars.rightBarLeaderboard.length > 0) {
       return;
     }
 
-    let readerPubKey = "";
-    if (this.globalVars.loggedInUser) {
-      readerPubKey = this.globalVars.loggedInUser.PublicKeyBase58Check;
-    }
-    this.backendApi
-      .GetProfiles(
-        this.globalVars.localNode,
-        null /*PublicKeyBase58Check*/,
-        null /*Username*/,
-        null /*UsernamePrefix*/,
-        null /*Description*/,
-        BackendApiService.GET_PROFILES_ORDER_BY_INFLUENCER_COIN_PRICE /*Order by*/,
-        10 /*NumEntriesToReturn*/,
-        readerPubKey /*ReaderPublicKeyBase58Check*/,
-        "leaderboard" /*ModerationType*/,
-        false /*FetchUsersThatHODL*/,
-        false /*AddGlobalFeedBool*/
-      )
-      .subscribe(
-        (response) => {
-          this.globalVars.rightBarLeaderboard = response.ProfilesFound.slice(
-            0,
-            RightBarCreatorsLeaderboardComponent.MAX_PROFILE_ENTRIES
-          );
+    const pulseService = new PulseService(this.httpClient, this.backendApi, this.globalVars);
+
+    pulseService.getBitCloutLockedLeaderboard().subscribe((res) => {
+      pulseService.getProfilesForPulseLeaderboard(res, PulseLeaderboardType.BitCloutLocked).subscribe({
```

Comment:
the response from `getBitCloutLockedLeaderboard` should include the profiles fetched from GetUsersStateless by chaining promises together. there are examples of this in backend service.